test/unittest/tests/backup/test_incremental_blkmods.cpp

```diff
@@ -0,0 +1,336 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include <catch2/catch.hpp>
+#include <iostream>
+#include <regex>
+#include <sstream>
+#include "wiredtiger.h"
+#include "wt_internal.h"
+#include "extern.h"
+#include "../utils.h"
+#include "../wrappers/connection_wrapper.h"
+#include "../wrappers/item_wrapper.h"
+
+static int
+insert_key_value(WT_CURSOR *cursor, const char *key, const char *value)
+{
+    item_wrapper item_key(key);
+    item_wrapper item_value(value);
+    __wt_cursor_set_raw_key(cursor, item_key.get_item());
+    __wt_cursor_set_raw_value(cursor, item_value.get_item());
+    return cursor->insert(cursor);
+}
+
+static void
+insert_key_value(
+  WT_CURSOR *cursor1, WT_CURSOR *cursor2, std::string const &key, std::string const &value)
+{
+    insert_key_value(cursor1, key.c_str(), value.c_str());
+    insert_key_value(cursor2, key.c_str(), value.c_str());
+}
+
+static void
+insert_sample_values(WT_CURSOR *cursor1, WT_CURSOR *cursor2, int first_value, int num_values)
+{
+    for (int loop = 0; loop <= num_values; loop++) {
+        int i = first_value + loop;
+        std::stringstream key;
+        key << "key" << i;
+        std::stringstream value;
+        value << "value" << i;
+        insert_key_value(cursor1, cursor2, key.str(), value.str());
+    }
+}
+
+static std::string
+parse_blkmods(WT_SESSION *session, std::string const &file_uri)
+{
+    WT_CURSOR *metadata_cursor = nullptr;
+    REQUIRE(session->open_cursor(session, "metadata:", nullptr, nullptr, &metadata_cursor) == 0);
+
+    metadata_cursor->set_key(metadata_cursor, file_uri.c_str());
+    REQUIRE(metadata_cursor->search(metadata_cursor) == 0);
+
+    char *file_config;
+    REQUIRE(metadata_cursor->get_value(metadata_cursor, &file_config) == 0);
+
+    std::cmatch match_results;
+    REQUIRE(std::regex_search(file_config, match_results, std::regex(",blocks=(\\w+)")));
+    std::string hex_blkmod = match_results[1];
+
+    REQUIRE(metadata_cursor->close(metadata_cursor) == 0);
+
+    return hex_blkmod;
+}
+
+static uint8_t
+get_hex_value_from_string(std::string const &source_string, size_t index)
```

Comment:
Have you considered a single function for transforming the string into a byte vector?
As I see it one of the downsides of this function is that is that if there is a non-hex digit(s) in the string it will not be discovered (if at all) until queried. Compiling the entire vector up front provides a single point of failure.

Additionally if an invalid index is passed to this function it returns zero rather than signalling an error via an exception -- is this intended? If so why is an invalid index not considered an error?

Reply:
The case where the function returned 0 was correct, as that's because no data can be interpreted as 0. However, I agree that is was unclear. 

I've following your suggestion of revamping the code to convert the hex strings up front and to make the invalid character and different string length cases clearer in the code and comments.